tutorials/scripting/gdscript/static_typing.rst

```diff
@@ -396,6 +396,77 @@ if you prefer a more readable and reliable, but more verbose syntax.
 ``UNSAFE_*`` warnings make unsafe operations more noticeable, than unsafe lines.
 Currently, ``UNSAFE_*`` warnings do not cover all cases that unsafe lines cover.
 
+Common unsafe operations and their safe counterparts
+----------------------------------------------------
+
+``UNSAFE_PROPERTY_ACCESS`` and ``UNSAFE_METHOD_ACCESS`` warnings
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+
+In this example, we aim to set a property and call a method on an object
+that has a script attached with ``class_name MyScript`` and that ``extends
+Node2D``. If we have a reference to the object as a ``Node2D`` (for instance,
+as it was passed to us by the physics system), we can first check if the
+property and method exist and then set and call them if they do::
+    
+    if "some_property" in node_2d:
+        node_2d.some_property = 20  # Produces UNSAFE_PROPERTY_ACCESS warning.
+    
+    if node_2d.has_method("some_function"):
+        node_2d.some_function()  # Produces UNSAFE_METHOD_ACCESS warning.
+
+However, this code will produce ``UNSAFE_PROPERTY_ACCESS`` and
+``UNSAFE_METHOD_ACCESS`` warnings as the property and method are not present
+in the referenced type - in this case a ``Node2D``. To make these operations
+safe, you can first check if the object is of type ``MyScript`` using the
+``is`` keyword and then declare a variable with the type ``MyScript`` on
+which you can set its properties and call its methods::
+    
+    if node_2d is MyScript:
+        var my_script: MyScript = node_2d
+        my_script.some_property = 20
+        my_script.some_function()
+
+Alternatively, you can use the ``as`` operator::
+
+    var my_script := node_2d as MyScript
+    if my_script:
```

Comment:
@dalexeev sorry, last question! In my opinion, `if my_script:` reads as if my_script is a boolean. I know that it actually means `if my_script != null:` so I was thinking to write that out like that in this example. Agreed?

Reply:
It's a bit complicated since in addition to `null` there is `Object#null` and `Freed Object` (see also godotengine/godot#73896), but I think that comparison with `null` will work here. There is also `is_instance_valid()`. On the other hand, implicit casting to `bool` has the advantage of fewer opcodes.

Reply:
Hmm, complicated indeed. Perhaps the most logical way of thinking about this particular case is that `my_script` is explicitly set to `null` if `node_2d` is not of type `MyScript`, and the explicitly verbose `null` check would help illustrate what happens when the cast fails. So even though I agree the fewer opcodes makes sense in most situations, perhaps in this example the slightly more verbose comparison to `null` makes sense in this example? I've pushed a new commit.